### ISSUE TITLE
The number of circuits on rides isn't save correctly

### ISSUE THREAD

**Issue by LordMarcel** [OpenRCT2/OpenRCT2#7304]
**OS:** Windows 10
**Version:** 0.1.2
**Commit/Build:** e70d201

In the vanilla, you could only have up to 7 circuits, but in open that was changed to 20. However, the number of circuits saved isn't the actual number of laps, but rather the number of laps modulo 8. If you save with 20 laps and rebuild it, you get 4, if you save with 8 laps you get 1 (actually 0, but you can't have 0 laps so it's changed to 1), etc.

**Comment by deurklink** [comment]
Hmm i quickly scanned for all places in the code where "circuits" is mentioned, and i saw the variable gets left-shifted in a few places. Maybe that causes some bits to get dropped somewhere?

**Comment by Gymnasiast** [comment]
This only applies to saving as a TD6 file, right?

**Comment by deurklink** [comment]
Found this in `static rct_track_td6 *track_design_save_to_td6(uint8 rideIndex)` : 

```
    td6->lift_hill_speed_num_circuits =
        ride->lift_hill_speed |
        (ride->num_circuits << 5);
```

**Comment by LordMarcel** [comment]
I wasn't very clear in the original post, my bad. The bug occurs when you save a track and then build the saved design. The ride you built from the saved design will have the incorrect number of circuits.

**Comment by Gymnasiast** [comment]
Yup, exactly what I thought. It's a TD6 limitation. I don't know when we will introduce a new track design format - could be before or after moving to a new save format.
